product_lifecycle/models/product.py

```diff
@@ -40,15 +40,19 @@ def write(self, values):
         """ Try to set the lifecycle_state to obsolete when quantity on hand will change
         the product to end instead.
         """
+        copy_vals = values.copy()
         for product in self:
             lc_state = values.get('lifecycle_state', product.lifecycle_state)
             available = values.get('virtual_available', product.virtual_available)
 
             if lc_state == 'obsolete' and available:
-                values.update({'lifecycle_state': 'end'})
+                copy_vals.update({'lifecycle_state': 'end'})
+                super(ProductProduct, product).write(copy_vals)
             elif lc_state == 'end' and not available:
-                values.update({'lifecycle_state': 'obsolete'})
-            super(ProductProduct, product).write(values)
+                copy_vals.update({'lifecycle_state': 'obsolete'})
+                super(ProductProduct, product).write(copy_vals)
+            else:
+                super(ProductProduct, product).write(values)
         return True
```

Comment:
Since you're calling the super method, you should return its result

Reply:
No, because it's a for

Reply:
You're right.